### PR TITLE
Update spec2termux.pl

### PR DESCRIPTION
**Pull request by qiongwu1016**

### CODE REVIEW

File: spec2termux.pl [hunk 1]
```diff
@@ -1 +1,580 @@
...
+
+        $$val_ref .= expand_macros $_;
+        next;
+    }
+
+    chomp;
+    die "unknown line: $_";
+}
+
+print Dumper($main_pkg);
```
Comment (layesuen) [inline spec2termux.pl]: rm

File: spec2termux.pl [hunk 2]
```diff
@@ -1 +1,580 @@
...
+safe_cd 'termux-scripts';
+
+if (! -d $name) {
+    safe_mkdir $name;
+}
+
+safe_cd $name;
+
+#write the build.sh file
+{
```
Comment (layesuen) [inline spec2termux.pl:264]: put these in a function

File: spec2termux.pl [hunk 3]
```diff
@@ -1 +1,580 @@
...
+        $license = $license_mapping{$license};
+        print "Mapped License: $license\n";
+    }
+    print $out <<_EOC_;
+#!/bin/bash
+set -x
+TERMUX_PKG_HOMEPAGE="$main_pkg->{URL}"
+TERMUX_PKG_DESCRIPTION="$main_pkg->{description}"
+TERMUX_PKG_LICENSE="$license"
+TERMUX_PKG_MAINTAINER="qiong.wu\@openresty.com"
```
Comment (layesuen) [inline spec2termux.pl:277]: make this an command line option

File: spec2termux.pl [hunk 4]
```diff
@@ -1 +1,580 @@
...
+TERMUX_PKG_VERSION="$main_pkg->{Version}"
+TERMUX_PKG_REVISION=2
```
Comment (layesuen) [inline spec2termux.pl:279]: should this be an option too?

File: spec2termux.pl [hunk 5]
```diff
@@ -1 +1,580 @@
...
+safe_cd $name;
+
+#write the build.sh file
+{
+    my $out = new_file 'build.sh';
+    $main_pkg->{description} =~ s/\n+/ /g;
+    my $license = $main_pkg->{License};
+    if (exists $license_mapping{$license}) {
+        $license = $license_mapping{$license};
+        print "Mapped License: $license\n";
```
Comment (layesuen) [inline spec2termux.pl]: rm debugging print

File: spec2termux.pl [hunk 6]
```diff
@@ -1 +1,580 @@
...
+                    print "\$dep: $dep\n";
+                	my ($tm_pkg, $tm_dep) = to_tm_pkg $dep;
+                    print "\$deb_pkg: $tm_pkg, \$deb_dep: $tm_dep\n";
+                	if ($tm_pkg ne '') {
+                    	push @tm_deps, $tm_dep;
+                    	push @tm_dep_pkgs, $tm_pkg;
+                	}
+        		}
+    		}	
+    	}
```
Comment (layesuen) [inline spec2termux.pl]: bad indentation

File: spec2termux.pl [hunk 7]
```diff
@@ -1 +1,580 @@
...
+            }
+        }
+
+        @tm_build_dep_pkgs = uniq @tm_build_dep_pkgs;
+        @tm_build_deps = uniq @tm_build_deps;
+        if (@tm_build_dep_pkgs) {
+            my $deps_str = join ', ', @tm_build_dep_pkgs;
+            print $out "TERMUX_PKG_BUILD_DEPENDS=\"$deps_str\"\n";
+        }
+	}
```
Comment (layesuen) [inline spec2termux.pl]: indentation

File: spec2termux.pl [hunk 8]
```diff
@@ -1 +1,580 @@
...
+    if (defined $install) {
+        $install =~ s/(\w+)='(.*?)(?<!\\)'/$1="$2"/gs; 
+        my $install_cmds = indent(amend_make_cmds($install), 4);
+        print $out <<"_EOC_";
+termux_step_make_install() {
+$install_cmds
+}
+_EOC_
+    }
+
```
Comment (layesuen) [inline spec2termux.pl]: extra newline

File: spec2termux.pl [hunk 9]
```diff
@@ -1 +1,580 @@
...
+    open(my $fh, '<:raw', $full_path) or die "Can't open '$full_path': $!";
+
+    my $sha256 = Digest::SHA->new(256);
+
+    $sha256->addfile($fh);
+
+    close($fh);
+
+    return $sha256->hexdigest;
```
Comment (layesuen) [inline spec2termux.pl]: why newlines

File: spec2termux.pl [hunk 10]
```diff
@@ -1 +1,580 @@
...
+
+sub amend_make_cmds($) {
+    my $make_cmds = shift;
+    $make_cmds = expand_macros($make_cmds);
+    $make_cmds =~ s/-C\s+src/-C \$TERMUX_PKG_SRCDIR\/src/g;
+    $make_cmds =~ s/(?<=make )(?!\-C)/-C \${TERMUX_PKG_SRCDIR} /g;
+    return $make_cmds;
+}
+
+sub gen_tarball_sha256 ($$) {
```
Comment (layesuen) [inline spec2termux.pl:397]: why space

File: spec2termux.pl [hunk 11]
```diff
@@ -1 +1,580 @@
...
+    my $sha256 = Digest::SHA->new(256);
+
+    $sha256->addfile($fh);
+
+    close($fh);
+
+    return $sha256->hexdigest;
+}
+
+sub gen_tm_pkg_name ($) {
```
Comment (layesuen) [inline spec2termux.pl:408]: rm space

File: spec2termux.pl [hunk 12]
```diff
@@ -1 +1,580 @@
...
+
+    } else {
+        $pkg_name = "$name-$key";
+        $pkg_name =~ s/-devel$/-dev/;
+    }
+
+    return $pkg_name;
+}
+
+sub to_tm_pkg ($) {
```
Comment (layesuen) [inline spec2termux.pl:423]: .

File: spec2termux.pl [hunk 13]
```diff
@@ -1 +1,580 @@
...
+    if ($dep =~ s/([><=].*)/($1)/) {
+        # strip release numbers for deb deps
+        $dep =~ s/-\d+\)$/)/;
+    }
+    $dep =~ s/\(\s*=\s*/(>= /;
+    #warn "returning $pkg_name $dep";
+    return ($pkg_name, $dep);
+}
+
+sub write_file ($$) {
```
Comment (layesuen) [inline spec2termux.pl:453]: .

File: spec2termux.pl [hunk 14]
```diff
@@ -1 +1,580 @@
...
+    my ($fname, $content) = @_;
+
+    my $out = new_file $fname;
+    print $out $content;
+    close $out;
+}
+
+sub new_file ($) {
```
Comment (layesuen) [inline spec2termux.pl:461]: ,

File: spec2termux.pl (hunk 11 above)
Comment (qiongwu1016) [reply spec2termux.pl:408]: 推荐格式有空格？

**Pull request merged**